etcdctl: add page (#5006)
tldr-pages/tldr: pages/common/etcdctl.md

# etcdctl

> > CLI interface for interacting with etcd high performance key/value pair store.

Reviewer: ```suggestion
> CLI interface for interacting with etcd high performance key-value pair store.
```

> > CLI interface for interacting with etcd high performance key-value pair store.

Reviewer 2: Perhaps this will read better?
```suggestion
> CLI interface for interacting with `etcd`, a highly-available key-value pair store.
```

> > Etcd stores keys in a filesystem-like format, e.g. /my/key.

Reviewer: ```suggestion
> Etcd stores data in hierarchically organized directories, as in a standard filesystem.
```

> More information: <https://etcd.io/docs/latest/dev-guide/interacting_v3/>.

> - Get a key:

Reviewer: The output is the `{{key}}\n{{value}}`, so:
```suggestion
- Display the key-value pair:
```

> `etcdctl get {{/my/key}}`

Reviewer: Why the key with `/`?

Author: @navarroaxel You mean the prefix?  That's what etcd does...see the homepage (https://etcd.io).

>### Key-value storage
>
>Store data in hierarchically organized directories, as in a standard filesystem
>```
>/config
>  /database
>/feature-flags
>  /verbose-logging
>  /redesign
>```

Reviewer: Oh, OK! but the `/` is required? The examples in more information link uses: `Here is the command to set the value of key foo to bar` without any `/`.

Author: Good point @navarroaxel.  I suppose we can remove it.

Author: @navarroaxel done @ 3ceea82781a8d89520ceaf24720a208e0c3483af

Reviewer: Yeah, I think that will be better.

> - Set a key:

Reviewer: ```suggestion
- Store a key-value pair:
```

> - Display the key-value pair:

Reviewer 2: Perhaps this will read better?
```suggestion
- Display the value associated with a specified key:
```

`etcdctl put {{my/key}} {{my_value}}`

> - Delete a key:

Reviewer: ```suggestion
- Delete a key-value pair:
```

`etcdctl del {{my/key}}`

> - Set a key, reading the value from a file:

Reviewer: ```suggestion
- Store a key-value pair, reading the value from a file:
```
